Cannot build for `ucans32k1sic` due to CMSIS and Kconfig mismatch

**Describe the bug**
Following changes in #64978, support for NXP S32K1 devices is broken due to mismatch between CMSIS feature definitions in device headers and Kconfig symbols defined at SoC.

**To Reproduce**
`west build -b ucans32k1sic samples/hello_world/ -p`

**Expected behavior**
build is successful

**Impact**
`main` broken for S32K1 devices

**Logs and console output**
```
In file included from /home/user/zephyrproject/zephyr/modules/cmsis/./cmsis_core.h:10,
                 from /home/user/zephyrproject/zephyr/arch/arm/include/cortex_m/stack.h:23,
                 from /home/user/zephyrproject/zephyr/arch/arm/include/kernel_arch_data.h:28,
                 from /home/user/zephyrproject/zephyr/arch/arm/core/offsets/offsets_aarch32.c:29,
                 from /home/user/zephyrproject/zephyr/arch/arm/core/offsets/offsets.c:9:
/home/user/zephyrproject/zephyr/modules/cmsis/./cmsis_core_m.h:56:2: error: #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value"
   56 | #error "__ICACHE_PRESENT and CONFIG_CPU_HAS_ICACHE are not set to the same value"
      |  ^~~~~
/home/user/zephyrproject/zephyr/modules/cmsis/./cmsis_core_m.h:60:2: error: #error "__DCACHE_PRESENT and CONFIG_CPU_HAS_DCACHE are not set to the same value"
   60 | #error "__DCACHE_PRESENT and CONFIG_CPU_HAS_DCACHE are not set to the same value"
      |  ^~~~~
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.16.3
 - Commit SHA 3d3b287b7908625dc8356ac3532d9b6943edbb50

**Additional context**
Regression of #64978

--- a/soc/arm/nxp_s32/s32k1/Kconfig.soc
+++ b/soc/arm/nxp_s32/s32k1/Kconfig.soc
@@ -20,48 +20,36 @@ config SOC_S32K142
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 config SOC_S32K142W
 	bool "S32K142W"
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 config SOC_S32K144
 	bool "S32K144"
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 config SOC_S32K144W
 	bool "S32K144W"
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 config SOC_S32K146
 	bool "S32K146"
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 config SOC_S32K148
 	bool "S32K148"
 	select CPU_CORTEX_M4
 	select CPU_CORTEX_M_HAS_DWT
 	select CPU_HAS_FPU
-	select CPU_HAS_DCACHE
-	select CPU_HAS_ICACHE
 
 endchoice
 

